### PR TITLE
Added first half of MS2 tests, not depending on reference rasterizer

### PR DESCRIPTION
**Pull request by marcorperez**
Signed-off-by: Marco R. Pérez <marco.perez@mobica.com>

### CODE REVIEW

File: sdk/tests/deqp/functional/gles3/es3fPrimitiveRestartTests.js [hunk 1]
```diff
@@ -0,0 +1,724 @@
...
+    /**
+    * @param {number} indexNdx
+    * @return {number}
+    */
+    es3fPrimitiveRestartTests.PrimitiveRestartCase.prototype.getIndex = function(indexNdx) {
+        switch (this.m_indexType) {
+            case es3fPrimitiveRestartTests.IndexType.INDEX_UNSIGNED_BYTE:
+                return this.m_indicesUB[indexNdx]; //deUint32
+            case es3fPrimitiveRestartTests.IndexType.INDEX_UNSIGNED_SHORT:
+                return this.m_indicesUS[indexNdx]; //deUint32
```
Comment (kenrussell) [inline sdk/tests/deqp/functional/gles3/es3fPrimitiveRestartTests.js]: Should this be deUint16?

File: sdk/tests/deqp/functional/gles3/es3fPrimitiveRestartTests.js [hunk 2]
```diff
@@ -0,0 +1,724 @@
...
+        }
+    };
+
+    /**
+    * Pointer to the index value at index indexNdx.
+    * @param {number} indexNdx
+    * @return {Uint8Array|Uint16Array|Uint32Array}
+    */
+    es3fPrimitiveRestartTests.PrimitiveRestartCase.prototype.getIndexPtr = function(indexNdx) {
+        //TODO: implement
```
Comment (kenrussell) [inline sdk/tests/deqp/functional/gles3/es3fPrimitiveRestartTests.js]: Is this TODO finished now?

**Comment by kenrussell** [comment]
Excellent work. Thank you. A couple of trivial comments above. Merging now.

**Comment by marcorperez** [comment]
Thanks Ken!

Will address your github comments tomorrow, i need to verify.

Best regards!

Marco Pérez
Mobica Ltd
Sr. Software Engineer
Mob: +52 3317 897600 <%2B44%20844%20875%201848>
Skype: mpez.mobica
www.mobica.com
<http://www.google.com/url?q=http%3A%2F%2Fwww.mobica.com%2F&sa=D&sntz=1&usg=AFrqEzeV0bYtmjTBtTc7DNhewmcdbKR9VA>

Mobica is a provider of innovative, cutting-edge software engineering,
testing and consultancy services. Based in the UK, Poland, the USA and
Mexico, Mobica has a worldwide customer base and a proven track record in
delivering complex solutions to global leaders in a range of sectors
including automotive, mobile, semiconductor, finance, TV & broadcasting,
telecommunications, connected devices, marine and aviation.

File: sdk/tests/deqp/functional/gles3/es3fPrimitiveRestartTests.js (hunk 1 above)
Comment (marcorperez) [inline sdk/tests/deqp/functional/gles3/es3fPrimitiveRestartTests.js]: Starting from the refactoring of the code to use the closure library, we've been trying to remove those uint32/uint16 references, only leaving comments in some cases, while using plain number types.

Comment (kenrussell) [inline sdk/tests/deqp/functional/gles3/es3fPrimitiveRestartTests.js]: Understood - the comment just needs to be fixed here.

Comment (marcorperez) [inline sdk/tests/deqp/functional/gles3/es3fPrimitiveRestartTests.js]: Oh, right. Didn't notice that

File: sdk/tests/deqp/functional/gles3/es3fPrimitiveRestartTests.js (hunk 2 above)
Comment (marcorperez) [inline sdk/tests/deqp/functional/gles3/es3fPrimitiveRestartTests.js]: Yes, we forgot to remove those notes.

**Pull request merged**